rmtk/vulnerability/common/utils.py

```diff
@@ -735,15 +735,30 @@ def calculate_mean_fragility(gmrs,PDM,T,damping,IMT,damage_model,method):
     cumPDM = numpy.fliplr(numpy.cumsum(numpy.fliplr(PDM),axis=1))
     cumPDM = numpy.transpose(cumPDM)
     cumPDM = cumPDM/no_assets
-
+        
     logmeans = []
     logstddev = []
-    for iDS in range(len(damage_model['damage_states'])):
-        if method == 'least squares':
+    if method == 'least squares':
+        for iDS in range(len(damage_model['damage_states'])):
             solution, flag = optimize.leastsq(residual_lognormal_dist, [0.1, 0.6], args=(imls, cumPDM[iDS+1]))
             logmeans.append(solution[0])
             logstddev.append(solution[1])
             
+    elif method == 'max likelihood':
+        # the imls and the PDM are sorted to be used in the maximum likelihood approach
+        # the PDM should be a list of integers in this approach
+        I = numpy.argsort(imls, axis=0)
+        imls = numpy.array(imls)
+        imls = imls[I]
```

Comment:
Because I need to sort correspondingly the PDM vector, so I need the vector of indices I, do you agree?